recipes/ttp/meta.yaml

```diff
@@ -0,0 +1,65 @@
+{% set name = "ttp" %}
+{% set version = "0.9.5" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ttp-{{ version }}.tar.gz
+  sha256: 234414f4d3039d2d1cde09993f89f8db1b34d447f76c6a402555cefac2e59c4e
+
+build:
+  entry_points:
+    - ttp = ttp.ttp:cli_tool
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >=3.7,<4.0
+    - poetry-core >=1.0.0
+    - pip
+  run:
+    - python >=3.7,<4.0
+  run_constrained:
```

Comment:
This list of run_constraint will make it very hard to be installed. These are probably optional dependencies for e.g. docs. You should drop them from the recipe.

Reply:
Hello @xhochy,

I'm not familiar with conda packaging, so sorry if i make mistake. I have removed sphinx related package in "run_constrained". I have read the documentation about "run_constrained" parameter but it's not clear for me. Does theses dependancies will be installed when i install the "ttp" package or we have a system to install optional dependencies like pip have, eg: "pip install ttp" vs "pip install ttp[full]"

Regards

Reply:
run_constrainted is only for additional constraints that limit the version ranges of these packages if tgey were to be installed because od another package. They don't get auto-installed. For optional dependencies, you can have a look at the black-feedstock. You need to use multiple outputs for that. I would not recommend that here as it's probably unlikely that one wants the conda package to work on the documents and not work on that in the actual source tree.